File: 1260-shift-2d-grid/1260-shift-2d-grid.cpp
class Solution {
public:
    vector<vector<int>> shiftGrid(vector<vector<int>>& a, int k) {
        int n = a.size(), m = a[0].size();
        auto b = a;
        k %= n*m;
        for (int i=0; i<n; i++) {
            for (int j=0; j<m; j++) {
                auto x = i*m+j+k;
                x %= n*m;
                b[x/m][x%m] = a[i][j];
            }
        }
        return b;        
    }
};

File: 1260-shift-2d-grid/1260-shift-2d-grid.py
class Solution:
    def shiftGrid(self, grid: List[List[int]], k: int) -> List[List[int]]:
        n = len(grid)
        m = len(grid[0])
        a = []
        k %= n*m
        for row in grid:
            a = a + row
        a = a[-k:] + a[:-k]
        ret = []
        for i in range(n):
            ret.append(a[(i*m):(i+1)*m])
        return ret
        